src-wasm/src/midi.rs

```diff
@@ -1,19 +1,34 @@
-use std::future::Future;
-use std::pin::Pin;
+use crate::EventHandler;
+use saturn_backend::midi::instruments::to_instrument;
+use saturn_backend::midi::note::MidiNote;
 use saturn_backend::syscall::{MidiHandler, MidiRequest};
+use std::pin::Pin;
+use std::{future::Future, sync::Arc};
 
-pub struct WasmMidi { }
+pub struct WasmMidi {
+    pub events: Arc<EventHandler>,
+}
 
 impl MidiHandler for WasmMidi {
-    fn play(&mut self, _request: &MidiRequest, _sync: bool) {
-        todo!()
+    fn play(&mut self, request: &MidiRequest, _sync: bool) {
+        let Some(name) = to_instrument(request.instrument as usize) else {
+            return;
+        };
+        self.events.send_midi_play(MidiNote {
+            sync: _sync,
```

Comment:
Only comment is maybe rename `_sync` to `sync`, since it is used now.
```suggestion
            sync,
```